src/daemon/ca/certauth.rs

```diff
@@ -1898,6 +1932,108 @@ impl CertAuth {
     }
 }
 
+/// # BGPSec
+///
+impl CertAuth {
+    pub fn bgpsec_definitions_show(&self) -> BgpSecCsrInfoList {
+        self.bgpsec_defs.info_list()
+    }
+
+    /// Process BGPSec Definition updates
+    pub fn bgpsec_definitions_update(
+        &self,
+        updates: BgpSecDefinitionUpdates,
+        config: &Config,
+        signer: &KrillSigner,
+    ) -> KrillResult<Vec<CaEvt>> {
+        let mut res = vec![];
+
+        let (additions, removals) = updates.unpack();
+
+        // We keep a copy of the definitions so that we can:
+        // a. remove and then re-add definitions
+        // b. use the updated definitions to generate objects in
+        //    applicable RCs
+        //
+        // (note: actual modifications of self are done when the events are applied)
+        let mut definitions = self.bgpsec_defs.clone();
+
+        for key in removals {
+            if !definitions.remove(&key) {
+                return Err(Error::BgpSecDefinitionUnknown(self.handle.clone(), key));
+            } else {
+                definitions.remove(&key);
```

Comment:
oops, that's because it used to be:
```
if !definitions.has(&key) {
```

But that's needlessly inefficient. Made a mistake when changing the code. Now the second call (which was the only call before) has been removed.